File: clients/typescript/examples/collections-update-one-nested.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Update a document
(async function () {
  const result = await collection.updateOne(
    { _id: "101" },
    {
      $set: {
        color: "blue",
        "address.city": "Austin",
        "classes.2": "biology",
      },
    },
  );

  console.log(result);
})();

File: clients/go/examples/collections-update-one-nested.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/filter"
	"github.com/datastax/astra-db-go/v2/astra/options"
	"github.com/datastax/astra-db-go/v2/astra/update"
)

func main() {
	ctx := context.Background()

	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Update a document
	result, err := collection.UpdateOne(
		ctx,
		filter.Eq("_id", "101"),
		update.Coll().
			Set("color", "blue").
			Set("address.city", "Austin").
			Set("classes.2", "biology"),
	)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(result)
}
